### DISCUSSION TITLE
Are Popup and Child context menus possible?

### DISCUSSION THREAD

**Discussion by ndepend** [icsharpcode/ILSpy#2604] [Development]
We'd like to have a popup context menu with several child menus like we do in our Reflector plugin (see the **NDepend >** popup menu in the screenshot below):

![image](https://user-images.githubusercontent.com/511445/148546934-f35b4b43-546d-4774-8263-c7d84aff720e.png)

It looks like it is not feasible, am I right?

This would be useful for sophisticated plugins. For example [the ILSpy Reflexil plugin](https://github.com/sailro/Reflexil/releases) adds a dozen of context menus. Wouldn't it be preferable to have a single **Reflexil >** popup menu with its child menus?

![image](https://user-images.githubusercontent.com/511445/148547161-8c7ab14b-4014-4de4-adf0-cd44494926e7.png)

**Comment by siegfriedpammer** [comment]
No, currently this is not possible. @sailro would you too be interested in a new API to allow this?

**Reply by sailro** [reply]
Yes why not

**Reply by ndepend** [reply]
@siegfriedpammer if you do the API it'd be great to make it recursive: a parent menu can have child menus that can be parent themselves
